keras/backend/tensorflow_backend.py

```diff
@@ -1584,3 +1584,112 @@ def random_binomial(shape, p=0.0, dtype=_FLOATX, seed=None):
     return tf.select(tf.random_uniform(shape, dtype=dtype, seed=seed) <= p,
                      tf.ones(shape, dtype=dtype),
                      tf.zeros(shape, dtype=dtype))
+
+# CTC
+# tensorflow has a native implemenation, but it uses sparse tensors
+# and therefore requires a wrapper for Keras. The functions below convert
+# dense to sparse tensors and also wraps up the beam search code that is 
+# in tensorflow's CTC implementation
+
+def ctc_label_dense_to_sparse(labels, label_lengths):
+    # undocumented feature soon to be made public    
+    from tensorflow.python.ops import functional_ops
+    label_shape = tf.shape(labels)
+    num_batches_tns = tf.pack([label_shape[0]])
+    max_num_labels_tns = tf.pack([label_shape[1]])
+
+    def range_less_than(previous_state, current_input):
+        return tf.expand_dims(tf.range(label_shape[1]), 0) < current_input
+
+    init = tf.cast(tf.fill(max_num_labels_tns, 0), tf.bool)
+    dense_mask = functional_ops.scan(range_less_than, label_lengths, 
+                                     initializer=init, parallel_iterations=1)
+    dense_mask = dense_mask[:, 0, :]
```

Comment:
What is your TF version? Here you are taking a slice, which as far as I can tell breaks shape inference and causes the following code to fail. A workaround is to manually set the shape of the slice.

This may not be the case with the latest TF version. Unclear